File: python_programming_basics/nested_loops/e_06_cinema_tickets.py
student_tickets = 0
standard_tickets = 0
kid_tickets = 0
movie_name = input()

while movie_name != "Finish":
    free_seats = int(input())
    sold_tickets = 0

    for _ in range(free_seats):
        ticket = input()

        if ticket == "End":
            break

        elif ticket == "student":
            student_tickets += 1

        elif ticket == "standard":
            standard_tickets += 1

        elif ticket == "kid":
            kid_tickets += 1

        sold_tickets += 1

    hall_percentage = sold_tickets / free_seats * 100
    print(f"{movie_name} - {hall_percentage:.2f}% full.")

    movie_name = input()

total_sold_tickets = student_tickets + standard_tickets + kid_tickets
print(f"Total tickets: {total_sold_tickets}")
print(f"{student_tickets / total_sold_tickets * 100:.2f}% student tickets.")
print(f"{standard_tickets / total_sold_tickets * 100:.2f}% standard tickets.")
print(f"{kid_tickets / total_sold_tickets * 100:.2f}% kids tickets.")

File: JavaScript Basics/06. Nested Loops/e-06-Cinema-Tickets.js
function calculateTicketStatistics(input) {
    let index = 0;
    let totalTickets = 0;
    let studentTickets = 0;
    let standardTickets = 0;
    let kidTickets = 0;

    while (input[index] !== "Finish") {
        let movie = input[index];
        index++;
        let availableSeats = Number(input[index]);
        index++;

        let soldTicketsForMovie = 0;
        while (input[index] !== "End" && input[index] !== "Finish") {
            let ticketType = input[index];
            index++;
            
            soldTicketsForMovie++;
            totalTickets++;

            if (ticketType === "student") {
                studentTickets++;
            } else if (ticketType === "standard") {
                standardTickets++;
            } else if (ticketType === "kid") {
                kidTickets++;
            }

            if (soldTicketsForMovie >= availableSeats) {
                break;
            }
        }

        let occupancyRate = (soldTicketsForMovie / availableSeats * 100).toFixed(2);
        console.log(`${movie} - ${occupancyRate}% full.`);

        if (input[index] === "End") {
            index++;
        }
    }

    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${((studentTickets / totalTickets) * 100).toFixed(2)}% student tickets.`);
    console.log(`${((standardTickets / totalTickets) * 100).toFixed(2)}% standard tickets.`);
    console.log(`${((kidTickets / totalTickets) * 100).toFixed(2)}% kids tickets.`);
}


calculateTicketStatistics(input);
